### PR TITLE
DEP: `lsim2` deprecated in favor of `lsim`

### LINKED ISSUE #15755: DEP: absorb lsim2 into lsim
Non-uniform timestamps have been deprecated since #4675

### PR DESCRIPTION
**Pull request by FranzForstmayr**
#### Reference issue
Closes gh-15755

#### What does this implement/fix?
Deprecate lsim2, make sure all tests for lsim2 work for lsim too

### CODE REVIEW

**Comment by FranzForstmayr** [comment]
@h-vetinari 
Would you like to take a first look at the changes?

I combined the tests of both `lsim` and `lsim2` to check for differences. There was a minor issue when using `lsim2` for MISO system, which is fixed now.

There are a few minor differences:
- `lsim2` validates the shape of the input array before returning anything. If the array is full of zeros, `lsim` will return the correct result, even if the shape of `U` does not match. I would prefer to check the shapes beforehand, and raise a `ValueError`. The original behavior is misleading, as `lsim` works with `np.zeros(x,y)`, but `np.ones(x,y)` throws a `ValueError`. However, this would probably need a legacy arg?
- `lsim2` does not work with the old `np.matrix` classes, should I spend any more attention to this case?
- I deprecated `impulse2` and `step2` too, is this okay? I kept the original function call and did not migrate `impulse2` too use `lsim`.

File: scipy/signal/_ltisys.py
Comment (tupui) [inline scipy/signal/_ltisys.py:1888]: Missing here and bellow a `stacklevel` (usually 2)

File: scipy/signal/_ltisys.py
Comment (tupui) [inline scipy/signal/_ltisys.py:2373]: We need to catch explicitly the warning, same bellow.

File: scipy/signal/tests/test_ltisys.py
Comment (tupui) [inline scipy/signal/tests/test_ltisys.py:546]: We need to match the exact message.

File: scipy/signal/tests/test_ltisys.py
Comment (tupui) [inline scipy/signal/tests/test_ltisys.py:554]: We need to match the exact warning

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you @FranzForstmayr. A few things from a cursory look.

File: scipy/signal/_ltisys.py
Comment (h-vetinari) [inline scipy/signal/_ltisys.py:2378]: Can we not translate this already to using `lsim` directly? Or do we just feel there's no point because `impulse2` itself will be removed as well?

File: scipy/signal/tests/test_cont2discrete.py
Comment (h-vetinari) [inline scipy/signal/tests/test_cont2discrete.py:298]: So IIUC, we we're passing these kwargs to `odeint` before. The new method should be better(?), but by removing the possibility to pass in keywords, we'll hard-break anyone who currently uses `kwargs`. I think we'll have to add something to `lsim` to warn on unknown keywords (and deprecate them).

**Review by h-vetinari — DISMISSED** [review]
Sorry I didn't get around to this earlier. I realised the situation is even more confusing than I thought, but it looks like you ended up doing the right thing (deprecating `lsim2`) anyway.

Still, some questions & suggestions.

**Comment by h-vetinari** [comment]
Given the amount of confusion (some self-imposed, some by the intricacies of the situation), I'd prefer a review from @WarrenWeckesser for this as well, since he [rewrote](https://github.com/scipy/scipy/commit/b351b7402b2e770a6acf285038cfebc62693cd5e) `lsim2` 12 years already 12 years ago and [noted](https://github.com/scipy/scipy/pull/4675#issuecomment-92085150) in #4675:

> As the author of the work-around `step2()`, let me say "Thanks!" for working on this! Deprecating `lsim2()`, `step2()` and `impulse2()` in favor of improved `lsim()`, `step()` and `impulse()` functions is a nice improvement.

**Comment by ilayn** [comment]
`lsim` is basically the simulation engine that uses state space based increments and is not meant for solving ODEs with nonuniform time samples. For that you need to use a proper ODE solver. 

However the code in #4675 was actually solving an essential problem that was needed for `python-control` and later it has been rewritten again https://github.com/python-control/python-control/pull/454 using a version of linear system response that was in `harold`. There is a very nontrivial way to get the necessary sample frequency and the total simulation time for such uses coming mostly out of signal and control folks.

Just to get a step response, ODE solver might also miss certain dynamics or give erroneous results. Thus the ideal is to deprecate `lsim2` and use `lsim`.

File: scipy/signal/_ltisys.py
Comment (FranzForstmayr) [reply scipy/signal/_ltisys.py:2378]: `**kwargs` for `lsim2` are not supported in `lsim`, so I cannot pass them through. I could drop them silently, but this could lead to backward incompabilities I assume. 
I thought this is fine, as `impulse2` will be removed as well

File: scipy/signal/tests/test_cont2discrete.py
Comment (FranzForstmayr) [reply scipy/signal/tests/test_cont2discrete.py:298]: Is this necessary? The user has to adopt the code anyway and switch to the new function, so removing the `**kwargs` would be possible at the same step.

File: scipy/signal/tests/test_ltisys.py
Comment (FranzForstmayr) [reply scipy/signal/tests/test_ltisys.py:530]: They should be in the general class, I'll double check

File: scipy/signal/_ltisys.py
Comment (h-vetinari) [reply scipy/signal/_ltisys.py:2378]: Yeah, with removing `impulse2` I agree that this is acceptable.

File: scipy/signal/tests/test_cont2discrete.py
Comment (h-vetinari) [reply scipy/signal/tests/test_cont2discrete.py:298]: Yes. I think it would be good to add something to the `lsim2` docstring about how to move such cases (that are using kwargs) to something equivalent with `lsim`

**Comment by FranzForstmayr** [comment]
@h-vetinari 
Sphinx build fails due to `DeprecationWarning`.
Should I filter the warning like in `step2` when calling `lsim2` or is there a better solution?

File: scipy/signal/_ltisys.py
Comment (h-vetinari) [inline scipy/signal/_ltisys.py:1899]:
```suggestion
    warnings.warn("lsim2 is deprecated and will be removed from scipy 1.12. "
                  "Use the feature-equivalent lsim function.",
```

**Review by h-vetinari — COMMENTED** [review]
Sorry for the delay in reviewing this. Looks good already, though now we need to reflect that this isn't going into 1.9 anymore.

**Comment by ilayn** [comment]
All warnings need a `stacklevel` argument

**Comment by h-vetinari** [comment]
Ping @FranzForstmayr :)

**Comment by FranzForstmayr** [comment]
@ilayn 
> All warnings need a `stacklevel` argument

Added a the stacklevel to `place_poles` function, as it was the only one in in this file without stacklevel.

File: scipy/signal/_ltisys.py
Comment (FranzForstmayr) [reply scipy/signal/_ltisys.py:1858]: At least I haven't found another method.

Comment (j-bowhay) [reply scipy/signal/_ltisys.py:1859]: @tupui when we did #16265 / #16273 which deprecated a function, it had to be removed to be removed from the refguid (the part I messed up :laughing:). That meant the example did not need to filter the warnings. Is this not the same scenario?

Comment (h-vetinari) [reply scipy/signal/_ltisys.py:1859]: Gentle ping @ev-br

**Comment by j-bowhay** [comment]
I think there is now a path forward with regard to the filtering of deprecation warnings for this thanks to #16866.
I believe it is now just a case of adding the function to https://github.com/scipy/scipy/blob/c85250e53c9fe0250ed8c91ef6fb7f6cfef548d3/doc/source/conf.py#L301-L310

**Comment by AnirudhDagar** [comment]
Yes @j-bowhay that should fix the issue here, I forgot to ping you after fixing the problem.

**Comment by FranzForstmayr** [comment]
Thank you @j-bowhay, this simplifies the example code.
However I have no clue what's the issue with the CI tests.

**Comment by AnirudhDagar** [comment]
@FranzForstmayr the issue is on [this line](https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=20239&view=logs&j=ede89041-5e8e-5201-ce5c-f68c2cbf02dd&t=bf6d2fae-c4b4-5aaf-e29f-d3092465da33&l=251). Thanks to @rgommers it will be fixed in #16906. You may ignore it for this PR.

**Review by h-vetinari — COMMENTED** [review]
Haven't reviewed in-depth yet, but from a first glance this LGTM (assuming CI passes)

Edit: I did start looking right after the last commit, not in the 5min since the review request 🙃

**Comment by tylerjereddy** [comment]
@h-vetinari @tupui thoughts here? I see a failing CI and a pretty complex discussion above. I think what I will do, given we are about 10 days from branching, is bump the milestone, since to be fair a deprecation isn't usually a "release blocker," but that isn't a statement on the value of the PR and I do hope this moves forward soon.

As always, if it goes get merged before branching, you can of course add the milestone back.

**Comment by h-vetinari** [comment]
To my mind we were able to resolve most discussions (my last comment was basically LGTM).

@FranzForstmayr, would you be able to merge master one last time and then we get this in? Sorry for all the delays on this.

**Comment by j-bowhay** [comment]
Agree, I think is looking in good shape now

File: scipy/signal/_ltisys.py
Comment (h-vetinari) [inline scipy/signal/_ltisys.py:1948]: Why is this change necessary?

File: scipy/signal/tests/test_ltisys.py
Comment (h-vetinari) [inline scipy/signal/tests/test_ltisys.py:416]: It seems you've taking this from the signature of `assert_almost_equal`, so I understand it this comes from. Still, I find
```
assert_almost_equal(x, expected_x, decimal=self.decimal)
```
quite jarring - for me, decimal refers primarily to a type (python, java, various SQL flavours).

So could we rename this? I don't have particularly good names in mind, but maybe one of
* `accuracy_digits`
* `digits_accuracy`
* `digits_of_accuracy`

File: scipy/signal/_ltisys.py
Comment (h-vetinari) [inline scipy/signal/_ltisys.py:1842]: I might be missing something, but it seems you're not using this in the docstring - does it fail if it's not there? 🤔

**Review by h-vetinari — COMMENTED** [review]
Thanks @FranzForstmayr for updating the PR ❤️ 

@tupui @j-bowhay 
I'd like to get this in before 1.10 branches, would you be able to take another look? Especially @tupui because you have some pending (though likely stale) requested changes.

All I have is nits, but (assuming the CI passes) this basically LGTM

File: scipy/signal/_ltisys.py
Comment (FranzForstmayr) [reply scipy/signal/_ltisys.py:1946]: Without this change `test_two_states` and `test_miso` fails.

"ValueError: shapes (2,2) and (1,2) not aligned: 2 (dim 1) != 1 (dim 0)" in `dot` line 1946

**Comment by FranzForstmayr** [comment]
@h-vetinari I needed to change some docstring examples in scipy.special and wavelet to get the sphinx build working again. I don't know, why this error is triggered in this PR and not before. I ran the example in scipy 1.9 and got the same warnings/errors as in the docbuild.

**Comment by h-vetinari** [comment]
Sorry we didn't make it for 1.10, thanks for sticking with this!

> @h-vetinari I needed to change some docstring examples in scipy.special and wavelet to get the sphinx build working again. I don't know, why this error is triggered in this PR and not before. I ran the example in scipy 1.9 and got the same warnings/errors as in the docbuild.

That's not the right way to do this unfortunately. Best case would have been to fix those warnings in the PRs that caused them, but barring that, could you open a PR just with 
https://github.com/scipy/scipy/pull/15929/commits/d6de9eca07501c6fb3a3b46b55ca30710e3d731c (plus the linter follow-up)?

Not all the changes are evident to me, and this should be discussed with the respective maintainers (to accelerate this, we could even split up the fixes into several PRs) - though since this is going to affect everyone, I'm guessing it'll be fixed soon even if you don't do anything.

File: scipy/signal/_ltisys.py
Comment (j-bowhay) [reply scipy/signal/_ltisys.py:1946]: Did this test not pass before? I would like to be clear on what make this change necessary before we proceed

Comment (FranzForstmayr) [reply scipy/signal/_ltisys.py:1946]: The original `test_miso` passed a 1d array for `u` into lsim, although the system consists of two inputs. I changed the test to be compliant to the documentation, then the test failed.

**Comment by ilayn** [comment]
Nevermind about the special and other lib errors. We can address them somewhere else. It would conflict with other PRs if we do it here, let's keep it specific to lsim related stuff. Having errors is not a problem we can live with non-green CI results.

**Comment by FranzForstmayr** [comment]
ruff won't let me commit the `doc/source/conf.py` file, as the special variable tags is not defined and ignoring the error for this line does not work.
See https://github.com/charliermarsh/ruff/issues/2561

**Comment by stefanv** [comment]
As mentioned in that issue, it's `# noqa:...` (note the space). Try that, and I'll take another look if it does not work.

**Comment by stefanv** [comment]
You can also add the conf file to ruff.toml under excludes so that file doesn't get checked, or under per-file-ignores for the specific error.

**Comment by FranzForstmayr** [comment]
@h-vetinari I changed the target version. CI is still failing, but I think not related to this PR as @ilayn mentioned

File: doc/source/conf.py
Comment (h-vetinari) [inline doc/source/conf.py:305]: OK, I see you need to (perhaps) add something here, but why all the other changes in this file?

I say "perhaps" because it seems to me that we don't need to add this to _every single code invocation_, but rather just those where one of the deprecated methods gets called, or better: change those call-sites.

**Review by h-vetinari — DISMISSED** [review]
Sorry that this is taking so many rounds (including changes "forced" upon you by the linter/CI), but this will still need some work I think. At least, I really don't think we should be changing so much of `doc/source/conf.py`, if anything at all.

File: doc/source/conf.py
Comment (j-bowhay) [reply doc/source/conf.py:305]: >  or better: change those call-sites.
I don't think this is possible in this circumstance. All the functions add here are the once being deprecated; it is impossible to avoid their own examples from generated a deprecation warning. I imagine the other changes relate to the recent changes in lint infrastructure

Comment (h-vetinari) [reply doc/source/conf.py:305]: > it is impossible to avoid their own examples from generated a deprecation warning.

We could remove their examples? Serious question

Comment (j-bowhay) [reply doc/source/conf.py:305]: Maybe? I am not sure it is worth it; they will get removed eventually anyway and there is no precedent or policy for doing so. I understand your hesitance about heavy handed filtering of warning but I think it is fine as it is only for the doc build, any unintended use of these deprecated functions would come out in the test suite.

Comment (FranzForstmayr) [reply doc/source/conf.py:305]: > OK, I see you need to (perhaps) add something here, but why all the other changes in this file?

The other changes were done due to the new linter configuration.

File: scipy/signal/_ltisys.py
Comment (h-vetinari) [inline scipy/signal/_ltisys.py:3411]:
```suggestion
                    f"You asked for a tolerance of {rtol}, we got {cur_tol}."
```

File: scipy/signal/_ltisys.py
Comment (h-vetinari) [inline scipy/signal/_ltisys.py:3410]:
```suggestion
                    f"You asked for a tolerance of {rtol}, we got {cur_rtol}."
```

**Review by h-vetinari — APPROVED** [review]
Did some very minor updates & merged main. LGTM, thanks for the work & your patience @FranzForstmayr!

@tupui, since you requested changes once upon a time, could you PTAL?

**Review by tupui — APPROVED** [review]
LGTM, thanks all. (I won't do some nitpicking on SciPy vs scipy in some messages as these are going to be removed soon either way.)

I relaunched the doc CI, it seemed unrelated but as we add some warnings, would be good to have a green CI there. EDIT: still failing but should not be related, I think it's just because we always have issues getting `main` merged properly on CircleCI. @h-vetinari your call as you reviewed this more closely.

**Comment by h-vetinari** [comment]
@FranzForstmayr, this is now just before the finish line. 🥳

However, the git history has gotten too convoluted for a normal merge, and in this case I would squash everything into one commit. If you'd like to rebase this to a streamlined history of a handful of sensible commits, you're welcome to do so, just let us know if you need some time for that.

And if you're fine with the squash, please just let us know as well. :)

If I don't hear anything else, I'm planning to (squash-)merge this in a week.

PS. Thanks a lot for the review @tupui! Agreed that the doc build shouldn't block here, as it looks unrelated.

**Comment by FranzForstmayr** [comment]
@h-vetinari 
I'm fine with a squash. Thanks for your time for reviewing :)

**Comment by mdhaber** [comment]
Looks like this caused doc build failures.

**Comment by h-vetinari** [comment]
Sorry for the breakage.

I've just checked out those warnings, and none of them are directly related to the deprecation in this PR...?

<details>

```
$SP_DIR/scipy/stats/_stats_py.py:docstring of scipy.stats._stats_py.kurtosistest:71: WARNING: Exception occurred in plotting scipy-stats-kurtosistest-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.stats.kurtosistest.rst:
Traceback (most recent call last):
  File "$SP_DIR/matplotlib/sphinxext/plot_directive.py", line 481, in _run_code
    exec(code, ns)
  File "<string>", line 12, in <module>
  File "$SP_DIR/scipy/stats/_stats_py.py", line 1806, in kurtosistest
    warnings.warn("kurtosistest only valid for n>=20 ... continuing "
UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=11
$SP_DIR/scipy/stats/_stats_py.py:docstring of scipy.stats._stats_py.normaltest:59: WARNING: Exception occurred in plotting scipy-stats-normaltest-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.stats.normaltest.rst:
Traceback (most recent call last):
  File "$SP_DIR/matplotlib/sphinxext/plot_directive.py", line 481, in _run_code
    exec(code, ns)
  File "<string>", line 12, in <module>
  File "$SP_DIR/scipy/stats/_stats_py.py", line 1993, in normaltest
    k, _ = kurtosistest(a, axis)
  File "$SP_DIR/scipy/stats/_stats_py.py", line 1806, in kurtosistest
    warnings.warn("kurtosistest only valid for n>=20 ... continuing "
UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=11
$SP_DIR/scipy/special/_basic.py:docstring of scipy.special._basic.y1p_zeros:40: WARNING: Exception occurred in plotting scipy-special-y1p_zeros-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.special.y1p_zeros.rst:
Traceback (most recent call last):
  File "$SP_DIR/matplotlib/sphinxext/plot_directive.py", line 481, in _run_code
    exec(code, ns)
  File "<string>", line 25, in <module>
  File "$SP_DIR/scipy/special/_basic.py", line 983, in yvp
    return _bessel_diff_formula(v, z, n, yv, -1)
  File "$SP_DIR/scipy/special/_basic.py", line 823, in _bessel_diff_formula
    s += p*L(v-n + i*2, z)
RuntimeWarning: invalid value encountered in add
$SP_DIR/scipy/special/_basic.py:docstring of scipy.special._basic.yvp:49: WARNING: Exception occurred in plotting scipy-special-yvp-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.special.yvp.rst:
Traceback (most recent call last):
  File "$SP_DIR/matplotlib/sphinxext/plot_directive.py", line 481, in _run_code
    exec(code, ns)
  File "<string>", line 32, in <module>
  File "$SP_DIR/scipy/special/_basic.py", line 983, in yvp
    return _bessel_diff_formula(v, z, n, yv, -1)
  File "$SP_DIR/scipy/special/_basic.py", line 823, in _bessel_diff_formula
    s += p*L(v-n + i*2, z)
RuntimeWarning: invalid value encountered in add
$SP_DIR/scipy/special/_basic.py:docstring of scipy.special._basic.jnyn_zeros:50: WARNING: Exception occurred in plotting scipy-special-jnyn_zeros-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.special.jnyn_zeros.rst:
Traceback (most recent call last):
  File "$SP_DIR/matplotlib/sphinxext/plot_directive.py", line 481, in _run_code
    exec(code, ns)
  File "<string>", line 23, in <module>
  File "$SP_DIR/scipy/special/_basic.py", line 983, in yvp
    return _bessel_diff_formula(v, z, n, yv, -1)
  File "$SP_DIR/scipy/special/_basic.py", line 823, in _bessel_diff_formula
    s += p*L(v-n + i*2, z)
RuntimeWarning: invalid value encountered in add
docstring of scipy.special.nbdtr:74: WARNING: Exception occurred in plotting scipy-special-nbdtr-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.special.nbdtr.rst:
Traceback (most recent call last):
  File "$SP_DIR/matplotlib/sphinxext/plot_directive.py", line 481, in _run_code
    exec(code, ns)
  File "<string>", line 28, in <module>
RuntimeWarning: floating point number truncated to an integer
docstring of scipy.special.nbdtrc:74: WARNING: Exception occurred in plotting scipy-special-nbdtrc-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.special.nbdtrc.rst:
Traceback (most recent call last):
  File "$SP_DIR/matplotlib/sphinxext/plot_directive.py", line 481, in _run_code
    exec(code, ns)
  File "<string>", line 28, in <module>
RuntimeWarning: floating point number truncated to an integer
$SP_DIR/scipy/signal/windows/_windows.py:docstring of scipy.signal.windows._windows.cosine:31: WARNING: Exception occurred in plotting scipy-signal-windows-cosine-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.signal.windows.cosine.rst:
Traceback (most recent call last):
  File "$SP_DIR/matplotlib/sphinxext/plot_directive.py", line 481, in _run_code
    exec(code, ns)
  File "<string>", line 17, in <module>
RuntimeWarning: divide by zero encountered in log10
$SP_DIR/scipy/signal/_wavelets.py:docstring of scipy.signal._wavelets.morlet:63: WARNING: Exception occurred in plotting scipy-signal-morlet-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.signal.morlet.rst:
Traceback (most recent call last):
  File "$SP_DIR/matplotlib/sphinxext/plot_directive.py", line 481, in _run_code
    exec(code, ns)
  File "<string>", line 8, in <module>
  File "$SP_DIR/matplotlib/pyplot.py", line 2812, in plot
    return gca().plot(
  File "$SP_DIR/matplotlib/axes/_axes.py", line 1690, in plot
    self.add_line(line)
  File "$SP_DIR/matplotlib/axes/_base.py", line 2304, in add_line
    self._update_line_limits(line)
  File "$SP_DIR/matplotlib/axes/_base.py", line 2327, in _update_line_limits
    path = line.get_path()
  File "$SP_DIR/matplotlib/lines.py", line 1029, in get_path
    self.recache()
  File "$SP_DIR/matplotlib/lines.py", line 662, in recache
    y = _to_unmasked_float_array(yconv).ravel()
  File "$SP_DIR/matplotlib/cbook/__init__.py", line 1335, in _to_unmasked_float_array
    return np.asarray(x, float)
numpy.ComplexWarning: Casting complex values to real discards the imaginary part
$SP_DIR/scipy/signal/windows/_windows.py:docstring of scipy.signal.windows._windows.bohman:28: WARNING: Exception occurred in plotting scipy-signal-windows-bohman-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.signal.windows.bohman.rst:
Traceback (most recent call last):
  File "$SP_DIR/matplotlib/sphinxext/plot_directive.py", line 481, in _run_code
    exec(code, ns)
  File "<string>", line 17, in <module>
RuntimeWarning: divide by zero encountered in log10
$SP_DIR/scipy/interpolate/_interpolate.py:docstring of scipy.interpolate._interpolate.interp2d:114: WARNING: Exception occurred in plotting scipy-interpolate-interp2d-1
 from /home/circleci/repo/doc/source/reference/generated/scipy.interpolate.interp2d.rst:
Traceback (most recent call last):
  File "$SP_DIR/matplotlib/sphinxext/plot_directive.py", line 481, in _run_code
    exec(code, ns)
  File "<string>", line 9, in <module>
  File "$SP_DIR/numpy/lib/utils.py", line 95, in newfunc
    warnings.warn(depdoc, DeprecationWarning, stacklevel=2)
DeprecationWarning: `interp2d` is deprecated!
`interp2d` is deprecated in SciPy 1.10 and will be removed in SciPy 1.12.0.

For legacy code, nearly bug-for-bug compatible replacements are
`RectBivariateSpline` on regular grids, and `bisplrep`/`bisplev` for
scattered 2D data.

In new code, for regular grids use `RegularGridInterpolator` instead.
For scattered data, prefer `LinearNDInterpolator` or
`CloughTocher2DInterpolator`.

For more details see
`https://gist.github.com/ev-br/8544371b40f414b7eaf3fe6217209bff`
looking for now-outdated files... none found
```

</details>

File: doc/source/conf.py
Comment (h-vetinari) [inline doc/source/conf.py:321]: The only global changes are here, though I don't see how this could break something, as we're _adding_ more warnings to filter.

**Review by h-vetinari — COMMENTED** [review]
I'm a bit baffled by how these (apparently pre-existing) warnings now break the doc build. At least I don't see anything else other than the warnings in the log.

But if we cannot determine the origin of the problem, I'm OK with reverting this PR for now.

**Comment by rgommers** [comment]
Sphinx works in mysterious ways:( But probably better to fix the problem by removing the `conf.py` changes (deleting the plotting from Examples that trigger the warnings in the deprecated functions would be fine) and/or fixing up the offending code.

**Comment by h-vetinari** [comment]
> But probably better to fix the problem by removing the `conf.py` changes (deleting the plotting from Examples that trigger the warnings in the deprecated functions would be fine) and/or fixing up the offending code.

I have a revert in https://github.com/scipy/scipy/pull/18471 that we hopefully don't need - indeed it fixes the doc issue. I've just opened an alternative based on the suggestion of removing the docstring examples, hopefully that works too: https://github.com/scipy/scipy/pull/18472

Fixing the offending code sounds like the sort of effort that shouldn't be rushed, as each "fix" would need to ensure that the docstring does not become wrong, and so need review from each owner. Should I open none, one or several issues for the sphinx warnings?

**Comment by rgommers** [comment]
> Should I open none, one or several issues for the sphinx warnings?

Not several I'd say, it's either none if the fix is already in the pipeline by someone, or one issue.

**Pull request merged**
